Objective-C/CBLReplicator.mm

```diff
@@ -502,10 +501,15 @@ - (void) c4StatusChanged: (C4ReplicatorStatus)c4Status {
         #if TARGET_OS_IPHONE
             [self endBackgrounding];
         #endif
-            if (_conflictCount > 0) {
-                CBLLogInfo(Sync, @"%@: Stopped but waiting for conflict resolution (pending = %d) "
-                                  "to finish before notifying.", self, _conflictCount);
-                _deferStoppedNotification = YES;
+        }
+        
+        // handle replicator status callback
+        if (_conflictCount > 0 && (c4Status.level == kC4Stopped || c4Status.level == kC4Idle)) {
+            CBLLogInfo(Sync, @"%@: Status = %d, but waiting for conflict resolution (pending = %d) "
+            "to finish before notifying.", self, c4Status.level, _conflictCount);
+            
+            _deferReplicatorNotification = YES;
```

Comment:
It's also possible that the IDLE status could be changed to Busy so the deferred IDLE might not valid anymore. For that case, we can reset _deferReplicatorNotification = NO.

Reply:
Please also review other scenarios of the status change from IDLE to something else to see the behavior of deferring the idle notification.